File: frontend/src/services/productService.js
import api from "./api";

// Get all products
export const getProducts = async () => {
  const response = await api.get("/products");
  return response.data;
};

// Get product by ID
export const getProductById = async (id) => {
  const response = await api.get(`/products/${id}`);
  return response.data;
};

// Search products by name
export const searchProducts = async (name) => {
  const response = await api.get("/products/search", {
    params: {
      name,
    },
  });

  return response.data;
};

// Get products by category
export const getProductsByCategory = async (category) => {
  const response = await api.get(`/products/category/${category}`);
  return response.data;
};

File: backend/product-service/src/main/java/com/onedaywear/product/service/ProductService.java
package com.onedaywear.product.service;

import java.util.List;
import org.springframework.web.multipart.MultipartFile;
import com.onedaywear.product.dto.ApiResponse;
import com.onedaywear.product.dto.ProductRequest;
import com.onedaywear.product.dto.ProductResponse;
import com.onedaywear.product.entity.Category;

public interface ProductService {

    ProductResponse addProduct(ProductRequest request);

    List<ProductResponse> getAllProducts();

    ProductResponse getProductById(Long id);

    ApiResponse deleteProduct(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);
    
    List<ProductResponse> searchProducts(String name);
    
    List<ProductResponse> getProductsByCategory(Category category);
    
    void decreaseStock(Long productId, Integer quantity);

    void increaseStock(Long productId, Integer quantity);
    
    Long getProductCount();
    
    String uploadProductImage(Long productId, MultipartFile file);
}